src/stash/src/upgrade/v18_to_v19.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::upgrade::MigrationAction;
+use crate::{StashError, Transaction, TypedCollection};
+
+pub mod objects_v18 {
+    include!(concat!(env!("OUT_DIR"), "/objects_v18.rs"));
+}
+pub mod objects_v19 {
+    include!(concat!(env!("OUT_DIR"), "/objects_v19.rs"));
+}
+
+/// Some Items have the incorrect schema namespace of User, this migration fixes them to have a
+/// namespace of System.
+pub async fn upgrade(tx: &'_ mut Transaction<'_>) -> Result<(), StashError> {
+    const ITEM_COLLECTION: TypedCollection<objects_v18::ItemKey, objects_v18::ItemValue> =
+        TypedCollection::new("item");
+
+    ITEM_COLLECTION
+        .migrate_to(tx, |entries| {
+            let mut updates = vec![];
+
+            for (item_key, item_value) in entries {
+                if let Some(gid) = &item_key.gid {
+                    // If the GlobalId is in the system namespace, the SchemaId should be too.
+                    if gid.is_system() {
+                        let new_key = objects_v19::ItemKey::from(item_key.clone());
+                        let mut new_value = objects_v19::ItemValue::from(item_value.clone());
+
+                        if let Some(schema_id) = new_value.schema_id {
```

Comment:
When would this ever be `None`?

Reply:
It shouldn't ever be `None`, but if it is, I would rather not panic in this migration

Reply:
For the values that should never be `None`, if we do end up seeing a `None`, then that probably means something has gone very wrong. The catalog may even be corrupted in some way. It might be better to loudly explode when we encounter that, rather than silently move on with a potentially invalid state.

Up to you though whether you think it's better to leave the panics out of this migration. I don't think it's a big deal either way.

Reply:
Great point! I updated the code to log an error to Sentry if we find a `None`, but don't panic just in case some downstream code handles this scenario already